src/drivers/warmup.ts

```diff
@@ -0,0 +1,129 @@
+import { createAction } from '@reduxjs/toolkit'
+import axios from 'axios-observable'
+import { Epic } from 'redux-observable'
+import { of, race, throwError } from 'rxjs'
+import { catchError, delay, filter, map, mapTo, mergeMap, withLatestFrom } from 'rxjs/operators'
+
+/**
+ * This driver handles cache warmups for a HTTP endpoint, in this case, specifically for v1 external adapters.
+ *
+ * This driver gets events that are sent in from v1 adapters, then tracks when each EA cache would get cold via configured TTL.
+ * If the cache is ever determined to turn cold, a warmup response is sent to that particular EA to keep its cache warm, avoiding
+ * performance penalties of a cache bust.
+ */
+
+interface WarmupRequestRecievedPayload {
+  /**
+   * The URL of the external adapter we should track cache warmup for
+   */
+  url: string
+  /**
+   * The body of data we should send when we send a warmup response to warm up a external adapter's cache
+   */
+  body: unknown
+  /**
+   * The time to live (TTL) in seconds of a warm up request. This number gets reset every single time
+   * a warmup request comes in from the same adapter, if it ever is allowed to expire, it means we should send
+   * a warm up response to the external adapter
+   */
+  ttl: number
+  /**
+   * Metadata related to the external adapter itself that we can index, ex. the current response data of the EA that it
+   * has received while warming up its cache
+   */
+  meta: any
+}
+
+/**
+ * These set of events are emitted when a v1 EA calls this service
+ */
+export const warmupRequestRecieved = createAction<WarmupRequestRecievedPayload>(
+  'WARMUP_REQUEST_RECIEVED',
+)
```

Comment:
`WARMUP_REQUEST_RECIEVED` is a CMD to start the warmup driver. I think we'd need two more events here.

- `WARMUP_ACTIVATED`
- `WARMUP_DEACTIVTED`

Let's say we see a `WARMUP_REQUEST_RECIEVED` for `/slow-aggregation-data`. We need to decide if we activate the warmup of that URL, or it's already active. If there is no warmup process active for that specific URL, we start it and trigger `WARMUP_ACTIVATED`.

We set up some TTL for the warmup process. If the `WARMUP_REQUEST_RECIEVED` is not seen for TTL, it expires, and the `WARMUP_DEACTIVTED` event is triggered.